core/tests/performance_tests/base.py

```diff
@@ -36,9 +37,22 @@ def _record_page_metrics_from_cached_session(self, page_url):
             self.data_fetcher.get_page_metrics_from_cached_session(page_url))
 
     def _record_page_timings_for_url(self, page_url):
-        self.page_metrics = (
-            self.data_fetcher.get_page_timings_for_url(page_url))
+        page_session_metrics = []
+
+        for _ in range(3):
+            page_session_metrics.append(
+                self.data_fetcher.get_page_timings_for_url(page_url))
+
+        self.page_metrics = perf_domain.MultiplePageSessionMetrics(
+            page_session_metrics)
 
     def _record_page_timings_from_cached_session(self, page_url):
-        self.page_metrics = (
-            self.data_fetcher.get_page_timings_from_cached_session(page_url))
+        page_session_metrics = []
+
+        for _ in range(3):
+            page_session_metrics.append(
+                self.data_fetcher.get_page_timings_from_cached_session(
+                    page_url))
+
+        self.page_metrics = perf_domain.MultiplePageSessionMetrics(
```

Comment:
Could also initialize ``page_metrics`` (here and elsewhere) like so:

```python
self.page_metrics = perf_domain.MultiplePageSessionMetrics(
    [
        self.data_fetcher.get_page_timings_from_cached_session(page_url)
     for _ in range(3)])
```

Up to you (an optional suggestion; I'm ambivalent over which you choose).